Add Docker and Docker Compose support

jason-ball opened the pull request:
I added documentation to the README

rasliche commented on Dockerfile:
```diff
@@ -0,0 +1,15 @@
+FROM node:alpine
+
+ENV NODE_ENV production
```
When I ran the `docker-compose build` command the first time, it failed and bailed. Nuxt had an error saying not to use this env variable so I commented this line out with `ENV NODE_ENV production` and it worked.

Any thoughts? I assume it worked for you otherwise you wouldn't have left it in there.

rasliche commented:
Does this container handle hot-reloading? How does it work for development? Other than my comment above about that environment variable, it started up no problem, so thank you for working on this. Did you test it on heroku yet?

jason-ball commented:
@rasliche I added another docker-compose file for production servers and made the main one for development. I added live reloading to the main docker-compose too. I'm still letting it build the full app when the container builds just to make sure there's no issues with node or docker itself. Also, you'll still need to rebuild the container if you `npm install` anything. I did that in case the app ever has any native node modules.

rasliche commented on docker-compose-prod.yml:
```diff
@@ -0,0 +1,40 @@
+      # CHANGE THESE!!!
+      MONGO_INITDB_ROOT_USERNAME: root
+      MONGO_INITDB_ROOT_PASSWORD: example
+      MONGO_INITDB_DATABASE: bluestar
+      MONGO_INITDB_USERNAME: bluestar
+      MONGO_INITDB_PASSWORD: example
+      MONGO_INITDB_DATABASE: bluestar
```
Are there duplicates in here?

rasliche commented on docker-compose.yml:
```diff
@@ -0,0 +1,43 @@
+      # CHANGE THESE!!!
+      MONGO_INITDB_ROOT_USERNAME: root
+      MONGO_INITDB_ROOT_PASSWORD: example
+      MONGO_INITDB_DATABASE: bluestar
+      MONGO_INITDB_USERNAME: bluestar
+      MONGO_INITDB_PASSWORD: example
+      MONGO_INITDB_DATABASE: bluestar
```
Same as -prod version, are these duplicates on purpose?

rasliche reviewed (commented):
Duplicate env variables in dockerfiles?
